tests/jlm/hls/ir/TriggerTypeTests.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 Nico Reißmann <nico.reissmann@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#include <test-registry.hpp>
+
+#include <jlm/hls/ir/hls.hpp>
+
+#include <cassert>
+
+static int
+TestComputeHash()
+{
+  using namespace jlm::hls;
+
+  // Arrange
+  triggertype triggerType1;
+  triggertype triggerType2;
+  jlm::llvm::iostatetype ioStateType;
+
+  // Act & Assert
+  assert(triggerType1.ComputeHash() == triggerType1.ComputeHash());
+  assert(triggerType2.ComputeHash() == triggerType2.ComputeHash());
+  assert(triggerType1.ComputeHash() != ioStateType.ComputeHash());
```

Comment:
It seems a bit off to assert that the hashes are strictly not the same, as hash collisions are possible. Since the hashes are based on `typeid().hash_code()` it might be the case that hash collisions are not possible (the standard recommends that implementations avoid this as much as possible)

Reply:
I agree with you, but I have no idea how to better check it. There is a chance for hash collisions, but it should hopefully be very unlikely. Before I introduced the type aspect to the hash for each implementation, we had guaranteed collisions for hashes from different types (which should be very unlikely and not guaranteed). These checks ensure that we will hopefully not introduce a bug again such that it becomes guaranteed.

Honestly, I have no idea of how to improve upon this. @caleridas Any idea how to improve from here (if it is even worth it).

Reply:
@haved @caleridas  I removed all the unit tests with the too strict asserts again.